setup.py

```diff
@@ -156,17 +157,23 @@ def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version, lib_s
             f"or ensure {lib_pkg_name}.pc is in PKG_CONFIG_PATH."
         )
 
+    crypto_cflags = list(cflags)  # copy the default cflags
```

Comment:
see line 44:
```
cflags = [
    '-Wall',
    '-Wextra',
    '-Wpointer-arith',
]
```

Reply:
It does not make sense to pass linker flags via CFLAGS. Linker seems to ignore them, and compiler complains.

```
cc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -g -fPIC -O2 -pipe -g -O2 -pipe -g -O2 -pipe -fPIC -Isrc/borg/crypto -I/usr/local/include/eopenssl11 -I/usr/local/include/python3.9 -c src/borg/crypto/_crypto_helpers.c -o /tmp/ports/pobj/borgbackup-1.3.0.pre20220414/borg-3ef355a98f882f525a4ac7fa58dd561003e0e70a/temp.openbsd-7.1-amd64-3.9/src/borg/crypto/_crypto_helpers.o -Wall -Wextra -Wpointer-arith -Wl,-rpath,/usr/local/lib/eopenssl11
cc: warning: -Wl,-rpath,/usr/local/lib/eopenssl11: 'linker' input unused [-Wunused-command-line-argument]
```
What do you think of the diff below?

```
--- setup.py
+++ setup.py
@@ -23,6 +23,7 @@
 import setup_docs
 
 is_win32 = sys.platform.startswith('win32')
+is_openbsd = sys.platform.startswith('openbsd')
 
 # Number of threads to use for cythonize, not used on windows
 cpu_threads = multiprocessing.cpu_count() if multiprocessing and multiprocessing.get_start_method() != 'spawn' else None
@@ -156,9 +157,15 @@ def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version, lib_s
             f"or ensure {lib_pkg_name}.pc is in PKG_CONFIG_PATH."
         )
 
+    crypto_ldflags = []
     if is_win32:
         crypto_ext_lib = lib_ext_kwargs(
             pc, 'BORG_OPENSSL_PREFIX', 'libcrypto', 'libcrypto', '>=1.1.1', lib_subdir='')
+    elif is_openbsd:
+        # use openssl (not libressl) because we need AES-OCB and CHACHA20-POLY1305 via EVP api
+        crypto_ext_lib = lib_ext_kwargs(
+            pc, 'BORG_OPENSSL_PREFIX', 'crypto', 'libecrypto11', '>=1.1.1')
+        crypto_ldflags += ['-Wl,-rpath=/usr/local/lib/eopenssl11']
     else:
         crypto_ext_lib = lib_ext_kwargs(
             pc, 'BORG_OPENSSL_PREFIX', 'crypto', 'libcrypto', '>=1.1.1')
@@ -167,6 +174,7 @@ def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version, lib_s
         dict(sources=[crypto_ll_source, crypto_helpers]),
         crypto_ext_lib,
         dict(extra_compile_args=cflags),
+        dict(extra_link_args=crypto_ldflags),
     )
 
     compress_ext_kwargs = members_appended(
```

Reply:
yeah, sorry, i overlooked this wasn't meant for cflags, but ldflags.